examples/hubert/train.py

```diff
@@ -0,0 +1,166 @@
+import pathlib
+from argparse import ArgumentParser
+from typing import Tuple
+
+from lightning import HuBERTPreTrainModule
+from pytorch_lightning import Trainer
+from pytorch_lightning.callbacks import ModelCheckpoint
+
+
+def run_train(args):
+    checkpoint_dir = args.exp_dir / f"checkpoints_{args.dataset}_{args.model_name}"
+    checkpoint = ModelCheckpoint(
+        checkpoint_dir,
+        monitor="Losses/val_loss",
+        mode="min",
+        save_top_k=5,
+        save_weights_only=True,
+        verbose=True,
+    )
+    train_checkpoint = ModelCheckpoint(
+        checkpoint_dir,
+        monitor="Losses/train_loss",
+        mode="min",
+        save_top_k=5,
+        save_weights_only=True,
+        verbose=True,
+    )
+    callbacks = [
+        checkpoint,
+        train_checkpoint,
+    ]
+    trainer = Trainer(
+        default_root_dir=args.exp_dir,
+        max_steps=args.max_updates,
+        num_nodes=args.num_nodes,
+        gpus=args.gpus,
+        accelerator="gpu",
+        strategy="ddp",
+        replace_sampler_ddp=False,
+        gradient_clip_val=args.clip_norm,
+        callbacks=callbacks,
+    )
+
+    model = HuBERTPreTrainModule(
+        model_name=args.model_name,
+        num_classes=args.num_classes,
+        dataset=args.dataset,
+        root_path=args.root_path,
+        feature_type=args.feature_type,
+        seconds_per_batch=args.seconds_per_batch,
+        learning_rate=args.learning_rate,
+        betas=args.betas,
+        eps=args.eps,
+        weight_decay=args.weight_decay,
+        warmup_updates=args.warmup_updates,
+        max_updates=args.max_updates,
+    )
+    trainer.fit(model)
+
+
+def cli_main():
+    parser = ArgumentParser()
```

Comment:
Since the are many arguments, adding default value manually is not future-proof.
It should use `ArgumentDefaultsHelpFormatter`. (see my previous comment)